proto/tsopb.proto

```diff
@@ -0,0 +1,128 @@
+syntax = "proto3";
+package tsopb;
+
+import "gogoproto/gogo.proto";
+import "rustproto.proto";
+
+option (gogoproto.sizer_all) = true;
+option (gogoproto.marshaler_all) = true;
+option (gogoproto.unmarshaler_all) = true;
+option (rustproto.lite_runtime_all) = true;
+
+option java_package = "org.tikv.kvproto";
+
+service TSO {
+    rpc Tso(stream TsoRequest) returns (stream TsoResponse) {}
+
+    rpc SyncMaxTS(SyncMaxTSRequest) returns (SyncMaxTSResponse) {}
+
+    rpc GetDCLocationInfo(GetDCLocationInfoRequest) returns (GetDCLocationInfoResponse) {}
+
+    rpc SetExternalTimestamp(SetExternalTimestampRequest) returns (SetExternalTimestampResponse) {}
+
+    rpc GetExternalTimestamp(GetExternalTimestampRequest) returns (GetExternalTimestampResponse) {}
+}
+
+// TenantID is the unique id of the tenant
+message TenantID {
+    // id is a sequntial id. 
+    uint32 id = 1;
+    // keyspace_group_id is the unique id of the keyspace to which the tenant belongs
+    uint32 keyspace_group_id = 2;
+}
+
+message RequestHeader {
+    // tenant_id is the unique id of the tenant
+    TenantID tenant_id = 1;
```

Comment:
Thought about setting it to optional, but decided not to do so. Let me know if you have different options.